### ISSUE TITLE
Release 6.3.0

### ISSUE THREAD

**Issue by AckslD** [cdown/clipmenu#213]
Hi! I wanted to update the version used on Nix packages to the latest version in `develop` branch but maybe it's nice to have that as a proper version. Would it make sense to merge develop into master and tag it with `6.3.0`? This would include for example the new commands of `clipctl`.

**Comment by cdown** [comment]
Hey! I'm actually about to release a complete rewrite in C which will be tagged as 7.0.0. I've been testing it over the last few days and I think it's more or less ready for external testing, after which it will be released.

**Comment by AckslD** [comment]
Cool, sounds great :)

**Comment by cdown** [comment]
The C rewrite is at 70cc5d2b6a2027ba186f1555c0e114c53dc7c578 -- feel free to try it out. Packaging shouldn't change much from how it is now, although I might add manual pages.

**Comment by AckslD** [comment]
Started trying it out @cdown, seems to all work as expected :+1: :)

**Comment by ferki** [comment]
@cdown: could you consider tagging the last commit of the Bash-based approach as a "final Bash release", please? I mean either 9735907061cc0a69734c887a35a048805539f5dc or 87e1641b50ff0cc57ca850c34fe4d4c8cbb60bb0.

It would also help me tremendously with packaging for Gentoo. I'd like to use the already available latest developments, and would be nice if I could bring it to all Gentoo users as well.

I believe using 6.3.0 may work well without interfering with 7.0.0 as the first release of the C-based rewrite.

**Comment by AckslD** [comment]
Hi @cdown, what's the current status of the C-based version (7.0.0?)? :) Anything that I can help/try to get this out? I also agree with @ferki about tagging 6.3.0 as well.

**Comment by cdown** [comment]
Clipmenu 7 is basically done :-) The final thing is #240, I think.

I will tag the final bash version before tagging 7, but note that it has not received the level of scrutiny of most releases.

**Comment by N-R-K** [comment]
@ferki I have a local snapshot ebuild based on commit eaf0d903 which I've been using for a while now (along with patching in #227 via `/etc/portage/patches`). I'll attach the ebuild here in case you're interested and/or to avoid duplicated efforts.

[clipmenu-6.2.0_p20241214.ebuild](https://github.com/user-attachments/files/18542094/clipmenu-6.2.0_p20241214.ebuild.txt)

I haven't sent a pull request for adding this ebuild to gentoo repos yet since clipmenu at it's current state has basically no documentation and thus not in a state to be packaged yet (IMO).

Also, this probably isn't the right place to be talking about gentoo ebuilds. So if you have any follow up or want to discuss anything feel free to send me a mail at `nrk AT disroot DOT org` to avoid clogging up this thread further :)

**Comment by N-R-K** [comment]
@cdown could we get a tagged release soon since all the blockers (?) for clipmenu 7 have been resolved now.

**Comment by cdown** [comment]
I believe all is done, but I am moving to Singapore in the coming weeks so I may be a bit in and out, and would like to test it until then. When we do the clipmenu 7 release, I'll also tag 6.3.0 (but again, bear in mind it will not have received much scrutiny).

**Comment by biopsin** [comment]
Clipnotify will also go depricated with release of Cm.7?

**Comment by N-R-K** [comment]
> [...] and would like to test it until then

I think it's been long enough of a testing period without any outstanding issues. Release would be nice (whenever you're free to do so of course, no pressure just a friendly reminder).

**Comment by biopsin** [comment]
Tested "v 7.0.0" and so far its rockin on voidlinux, but I have two questions;
- `touch_on_select=1` does not move the selected to newest slot and remains in same position.
- setting `selections="clipboard"` still logs selection from primary.

my clipmenu.conf
```
max_clips=200
max_clips_batch=100
oneshot=0
own_clipboard=0
own_selections="clipboard"
selections="clipboard"
ignore_window=
launcher="dmenu"
launcher_pass_dmenu_args=1
touch_on_select=1
partial_merge_secs=0
cm_dir=

```

**Comment by N-R-K** [comment]
> * `touch_on_select=1` does not move the selected to newest slot and remains in same position.

Works for me :tm:. Try using `CM_DEBUG=1` and attach the logs? Also this should be in a separate issue.

**Comment by cdown** [comment]
The config format just looks wrong. You can't use `=` or quotes in clipmenu.conf.

**Comment by N-R-K** [comment]
> The config format just looks wrong. You can't use `=` or quotes in clipmenu.conf.

Ah, right. Didn't even realize it. That's probably the issue then.

**Comment by biopsin** [comment]
> Ah, right. Didn't even realize it. That's probably the issue then.

yep it was :)  I thought equal and quotes was a thing as on first try I got some errors without.
